### ISSUE TITLE
Zipkin reporter silent even when URL is invalid (or service down)

### ISSUE THREAD

**Issue by yngwietiger** [spring-cloud/spring-cloud-sleuth#836]
Hi,

I'm on **Dalston.SR5**, and I notice that if I configure a valid Zipkin URL (e.g. spring.zipkin.baseUrl=http://localhost:9411/), sampling and reporting works fine, and I'll see the results in the Zipkin UI.

However, I notice that if I use an invalid url, or the service happens to be down, I see nothing in the logs to indicate a problem.

But, on **Dalston.SR4**, it used to throw an exception....

For example, in this case the Zipkin server is down:

```
11:47:25.812 [AsyncReporter(org.springframework.cloud.sleuth.zipkin.RestTemplateSender@9e62fb)]  WARN [screenshot,,,] z.r.AsyncReporter$BoundedAsyncReporter - Dropped 1 spans due to ResourceAccessException(I/O error on POST request for "http://localhost:9411/api/v1/spans": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)) (AsyncReporter.java:284) 
org.springframework.web.client.ResourceAccessException: I/O error on POST request for "http://localhost:9411/api/v1/spans": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:738)
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:699)
	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:657)
	at org.springframework.cloud.sleuth.zipkin.RestTemplateSender.post(RestTemplateSender.java:73)
	at org.springframework.cloud.sleuth.zipkin.RestTemplateSender.sendSpans(RestTemplateSender.java:46)
	at zipkin.reporter.AsyncReporter$BoundedAsyncReporter.flush(AsyncReporter.java:245)
	at zipkin.reporter.AsyncReporter$Builder.lambda$build$0(AsyncReporter.java:166)
	at zipkin.reporter.AsyncReporter$Builder$$Lambda$1.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at java.net.Socket.connect(Socket.java:538)
	at sun.net.NetworkClient.doConnect(NetworkClient.java:180)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
	at sun.net.www.http.HttpClient.<init>(HttpClient.java:242)
	at sun.net.www.http.HttpClient.New(HttpClient.java:339)
	at sun.net.www.http.HttpClient.New(HttpClient.java:357)
	at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1202)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1138)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1032)
	at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:966)
	at org.springframework.http.client.SimpleBufferingClientHttpRequest.executeInternal(SimpleBufferingClientHttpRequest.java:78)
	at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48)
	at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:53)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:724)
	... 8 common frames omitted
```
I'm assuming this change was not intentional? i.e. we'd want to know when Zipkin is down

And one follow-up question:

I've noticed that when trace headers are propagated into a downstream application (and X-B3-Sampled = 1), that zipkin will always sample/report calls related to that request (regardless of sample percentage), which makes sense. But is there any flag/configuration for sleuth/zipkin to turn off sampling/reporting completely?

Thanks in advance,

yngwietiger

**Comment by marcingrzejszczak** [comment]
You can set `spring.sleuth.sampler.percentage=0.0` and nothing will get sampled.

The reporter is async so things happen in a different thread. Also any problems related to sending spans we assume as non critical - that's why we're logging a warning and not throwing an exception. That's why you have metrics which you can observe.

**Comment by yngwietiger** [comment]
Regarding the sampler percentage...

As I mentioned above, even if I set the percentage to 0.0, it still samples/reports as long as the trace headers are propagated and the X-B3-Sampled flag = 1. (I assumed this was desired behavior. But that's definitely what I've observed with both Sleuth and Brave.)

I just verified this on my local machine.

Regarding logging...

Okay, per the README, I bumped the logging level down to DEBUG (--logging.level.zipkin=DEBUG --logging.level.zipkin2=DEBUG), and now I see the error messages. Thanks.
